pkg/controllers/vdb/dbaddnode_reconciler.go

```diff
@@ -155,18 +177,22 @@ func (d *DBAddNodeReconciler) runAddNode(ctx context.Context, pods []*PodFact) (
 
 // runAddNodeForPod will execute the command to add a single node to the cluster
 // Returns the stdout from the command.
-func (d *DBAddNodeReconciler) runAddNodeForPod(ctx context.Context, pods []*PodFact, initiatorPod *PodFact) error {
-	podNameStr := genPodNames(pods)
+func (d *DBAddNodeReconciler) runAddNodeForPod(ctx context.Context,
+	expectedNodeNames []string,
+	podsToAdd []*PodFact,
+	initiatorPod *PodFact) error {
+	podNameStr := genPodNames(podsToAdd)
 	d.VRec.Eventf(d.Vdb, corev1.EventTypeNormal, events.AddNodeStart,
 		"Starting add database node for pod(s) '%s'", podNameStr)
 	start := time.Now()
 	opts := []addnode.Option{
 		addnode.WithInitiator(initiatorPod.name, initiatorPod.podIP),
-		addnode.WithSubcluster(pods[0].subclusterName),
+		addnode.WithSubcluster(podsToAdd[0].subclusterName),
 	}
-	for i := range pods {
-		opts = append(opts, addnode.WithHost(pods[i].dnsName, pods[i].name))
+	for i := range podsToAdd {
+		opts = append(opts, addnode.WithHost(podsToAdd[i].dnsName, podsToAdd[i].name))
 	}
+	addnode.WithExpecteNodeNames(expectedNodeNames)
```

Comment:
Can we include this with the initial setup of opts? Just move this line up to about line 190.